src/dbus/xdg_deskop_portal_impl_access.rs

```diff
@@ -0,0 +1,150 @@
+use std::ops::Not;
+
+use zbus::dbus_interface;
+
+use super::Start;
+
+pub struct AccessPortalImpl {
+    to_niri: calloop::channel::Sender<AccessDialogRequest>,
+}
+
+#[dbus_interface(name = "org.freedesktop.impl.portal.Access")]
+impl AccessPortalImpl {
+    /// AccessDialog method
+    #[allow(clippy::too_many_arguments)]
+    async fn access_dialog(
+        &self,
+        _handle: zbus::zvariant::ObjectPath<'_>,
+        app_id: &str,
+        parent_window: &str,
+        title: &str,
+        subtitle: &str,
+        body: &str,
+        options: std::collections::HashMap<&str, zbus::zvariant::Value<'_>>,
+    ) -> zbus::fdo::Result<(
+        u32,
+        std::collections::HashMap<String, zbus::zvariant::OwnedValue>,
+    )> {
+        let (response_channel_sender, response_channel_receiver) = async_channel::bounded(1);
+
+        let options = AccessDialogOptions::from_options(options);
+        let request = AccessDialogRequest {
+            app_id: app_id.to_string(),
+            parent_window: parent_window
+                .is_empty()
+                .not()
+                .then(|| parent_window.to_string()),
+            title: title.to_string(),
+            subtitle: subtitle.to_string(),
+            body: body.is_empty().not().then(|| body.to_string()),
+            options,
+
+            response_channel_sender,
+        };
+
+        if let Err(err) = self.to_niri.send(request) {
+            tracing::warn!(?err, "failed to send access dialog request");
+            return Ok((2, Default::default()));
+        };
+
+        let result = std::collections::HashMap::<String, zbus::zvariant::OwnedValue>::new();
```

Comment:
Why not `use` the HashMap? Also zbus::zvariant::Something and tracing::warn

Reply:
I am not sure I get what you mean? The hashmap is not needed atm, it would only be used for returning the choices on a grant, so I could return it. I left it here as a guidance for adding choices.

Not sure what you want me to do regarding `zbus::zvariant::Something and tracing::warn`

Reply:
I mean just import them instead of using fully qualified paths.

Reply:
changed